Nautilus/Assets/PrefabTemplates/CloneTemplate.cs

```diff
@@ -35,24 +37,30 @@ public class CloneTemplate : PrefabTemplate
     /// </summary>
     /// <param name="info">The prefab info to base this template off of.</param>
     /// <param name="techTypeToClone">The tech type to clone and use for this template.</param>
-    public CloneTemplate(PrefabInfo info, TechType techTypeToClone) : this(info, techTypeToClone, null) {}
+    public CloneTemplate(PrefabInfo info, TechType techTypeToClone) : this(info, techTypeToClone, null, null) {}
     
     /// <summary>
     /// Creates a <see cref="CloneTemplate"/> instance.
     /// </summary>
     /// <param name="info">The prefab info to base this template off of.</param>
     /// <param name="classIdToClone">The class ID to clone and use for this template.</param>
-    public CloneTemplate(PrefabInfo info, string classIdToClone) : this(info, TechType.None, classIdToClone) {}
+    public CloneTemplate(PrefabInfo info, string classIdToClone) : this(info, TechType.None, classIdToClone, null) {}
+
+    /// <summary>
+    /// Creates a <see cref="CloneTemplate"/> instance.
+    /// </summary>
+    /// <param name="info">The prefab info to base this template off of.</param>
+    /// <param name="prefabToClone">The AssetReferenceGameObject with a valid key to clone and use for this template.</param>
+    public CloneTemplate(PrefabInfo info, AssetReferenceGameObject prefabToClone) : this(info, TechType.None, null, prefabToClone) { }
+
 
-    private CloneTemplate(PrefabInfo info, TechType techTypeToClone, string classIdToClone) : base(info)
+
+    private CloneTemplate(PrefabInfo info, TechType techTypeToClone, string classIdToClone, AssetReferenceGameObject prefabToClone) : base(info)
     {
         _techTypeToClone = techTypeToClone;
         _classIdToClone = classIdToClone;
-        _spawnType = techTypeToClone switch
-        {
-            default(TechType) => SpawnType.ClassId,
-            _ => SpawnType.TechType
-        };
+        _prefabToClone = prefabToClone;
+        _spawnType = techTypeToClone != TechType.None ? SpawnType.TechType : string.IsNullOrWhiteSpace(classIdToClone) ? prefabToClone != null ? SpawnType.Prefab : throw new System.Exception("CloneTemplate Missing valid identifier.") : SpawnType.ClassId;
```

Comment:
This is very hard to read, I suggest simplifying it and making it read like a stack instead:
```suggestion
        _spawnType = techTypeToClone is not TechType.None ? SpawnType.TechType
            : !string.IsNullOrWhiteSpace(classIdToClone) ? SpawnType.ClassId
            : prefabToClone is not null ? SpawnType.Prefab
            : throw new System.Exception("CloneTemplate Missing valid identifier.");
```